homeassistant/components/notify/telegram.py

```diff
@@ -63,52 +65,83 @@ def send_message(self, message="", **kwargs):
+                            auth=HTTPBasicAuth(photo_data[ATTR_USERNAME],
+                                               photo_data[ATTR_PASSWORD])
+                        )
+                    else:
+                        req = requests.get(photo_data[ATTR_URL])
+                    file_id = io.BytesIO(req.content)
+                elif ATTR_FILE in photo_data:
+                    file_id = open(photo_data[ATTR_FILE], "rb")
+                else:
+                    _LOGGER.error("No url or path is set for photo!")
+                    continue
+
+                self.bot.sendPhoto(chat_id=self._chat_id,
+                                   photo=file_id, caption=caption)
+
+        except (OSError, IOError, telegram.error.TelegramError,
+                urllib.error.HTTPError):
+            _LOGGER.exception("Error sending photo.")
             return
 
-        # send photo
-        if ATTR_PHOTO in data:
-            # if not a list
-            if not isinstance(data[ATTR_PHOTO], list):
-                photos = [data[ATTR_PHOTO]]
-            else:
-                photos = data[ATTR_PHOTO]
+    def send_location(self, data):
+        """Send a location to a user."""
+        import telegram
+
+        if ATTR_LOCATION in data:
+            location_entity = data[ATTR_LOCATION]
 
             try:
-                for photo_data in photos:
-                    caption = photo_data.get(ATTR_CAPTION, None)
-
-                    # file is a url
-                    if ATTR_URL in photo_data:
-                        # use http authenticate
-                        if ATTR_USERNAME in photo_data and\
-                           ATTR_PASSWORD in photo_data:
-                            req = requests.get(
-                                photo_data[ATTR_URL],
-                                auth=HTTPBasicAuth(photo_data[ATTR_USERNAME],
-                                                   photo_data[ATTR_PASSWORD])
-                            )
-                        else:
-                            req = requests.get(photo_data[ATTR_URL])
-                        file_id = io.BytesIO(req.content)
-                    elif ATTR_FILE in photo_data:
-                        file_id = open(photo_data[ATTR_FILE], "rb")
+                for location_data in location_entity:
+
+                    if ATTR_LONGITUDE in location_data and\
+                       ATTR_LATITUDE in location_data:
+                        latitude = location_data[ATTR_LATITUDE]
+                        longitude = location_data[ATTR_LONGITUDE]
                     else:
-                        _LOGGER.error("No url or path is set for photo!")
+                        _LOGGER.error("Long and Lat not set for location!")
                         continue
 
-                    self.bot.sendPhoto(chat_id=self._chat_id,
-                                       photo=file_id, caption=caption)
+                self.bot.sendLocation(chat_id=self._chat_id,
```

Comment:
Why would you iterate through the whole list and only sent the last latitude and longitude ?